namespace-registry/client_commands.go

```diff
@@ -0,0 +1,188 @@
+package nsregistry
+
+import (
+	"crypto/tls"
+
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"net/http"
+	"bytes"
+	"bufio"
+	"math/big"
+	"encoding/base64"
+
+	"github.com/pelicanplatform/pelican/config"
+	"github.com/spf13/viper"
+)
+
+func makeRequest(url string, method string, data map[string]interface{}) ([]byte, error) {
+	payload, _ := json.Marshal(data)
+	req, err := http.NewRequest(method, url, bytes.NewBuffer(payload))
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	if viper.GetBool("TLSSkipVerify") {
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+		client = &http.Client{Transport: tr}
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil,err
+	}
+	defer resp.Body.Close()
+
+	body, _ := ioutil.ReadAll(resp.Body)
+	return body, nil
+}
+
+func respToJSON(body []byte) (map[string]string) {
+	// Unmarshal the response body
+	var respData map[string]string
+	err := json.Unmarshal(body, &respData)
+	if err != nil {
+		panic(err)
+	}
+	return respData
+}
+
+func NamespaceRegisterWithIdentity(privateKeyPath string, namespaceRegistryEndpoint string, prefix string) (error) {
+	data := map[string]interface{}{
+		"identity_required": "true",
+	}
+	resp, err := makeRequest(namespaceRegistryEndpoint, "POST", data)
+	if err != nil {
+		return fmt.Errorf("Failed to make request: %s\n", err)
+	}
+	respData := respToJSON(resp)
```

Comment:
Switch to unmarshaling from a struct.  That way we can avoid some of the boilerplate type safety issues.